apprise/plugins/NotifyThreema.py

```diff
@@ -0,0 +1,368 @@
+        },
+        'target_user': {
+            'name': _('Target Email'),
+            'type': 'string',
+            'map_to': 'targets',
+        },
+        'targets': {
+            'name': _('Targets'),
+            'type': 'list:string',
+        },
+    })
+
+    # Define our template arguments
+    template_args = dict(NotifyBase.template_args, **{
+        'to': {
+            'alias_of': 'targets',
+        },
+        'from': {
+            'alias_of': 'user',
+        },
+        'secret': {
+            'alias_of': 'secret',
+        },
+    })
+
+    def __init__(self, secret=None, targets=None, **kwargs):
+        """
+        Initialize Threema Object
+        """
+        super().__init__(**kwargs)
+
+        # Validate our params here.
+
+        if not self.user:
+            msg = 'Threema Gateway UserID must be specified'
+            self.logger.warning(msg)
+            raise TypeError(msg)
+
+        if len(self.user) != 8:
+            msg = 'Threema Gateway UserID must be 8 characters in length'
+            self.logger.warning(msg)
+            raise TypeError(msg)
+
+        # We always use a token if provided
+        self.secret = validate_regex(secret)
+        if not self.secret:
+            msg = \
+                'An invalid Threema API Secret ({}) was specified'.format(
+                    secret)
+            self.logger.warning(msg)
+            raise TypeError(msg)
+
+        # Parse our targets
+        self.targets = list()
+
+        # Used for URL generation afterwards only
+        self.invalid_targets = list()
+
+        if targets:
+            for target in parse_phone_no(targets):
+                if len(target) == 8:
+                    # Store our user
+                    self.targets.append(
+                        (ThreemaRecipientTypes.USER, target))
+                    continue
+
+                # Check if an email was defined
+                result = is_email(target)
+                if result:
+                    # Store our user
+                    self.targets.append(
+                        (ThreemaRecipientTypes.EMAIL, result['full_email']))
+                    continue
+
+                # Validate targets and drop bad ones:
+                result = is_phone_no(target)
+                if result:
+                    # store valid phone number
+                    self.targets.append((
+                        ThreemaRecipientTypes.PHONE,
+                        '{}{}'.format(
+                            '' if target[0] != '+' else '+', result['full'])))
+                    continue
+
+                self.logger.warning(
+                    'Dropped invalid user/email/phone '
+                    '({}) specified'.format(target),
+                )
+                self.invalid_targets.append(target)
+
+        else:
+            # Send a message to ourselves
+            self.targets.append(
+                (ThreemaRecipientTypes.USER, self.user))
```

Comment:
As indicated by @threema-eduard this is not possible? Also even if it were, one would need to setup an API endpoint etc. for receiving Threema messages via the gateway API. (And that cannot be done with AppRise, so likely few people would use it.)

Reply:
yeah